pkg/forwarder/retry_queue_telemetry.go

```diff
@@ -5,126 +5,252 @@
+}
+
+type gaugeExpvar struct {
+	gauge  telemetry.Gauge
+	expvar expvar.Int
+}
+
+func newGaugeExpvar(subsystem string, name string, expvarName string, help string, parent *expvar.Map) *gaugeExpvar {
+	g := &gaugeExpvar{
+		gauge: telemetry.NewGauge(subsystem, name, []string{}, help),
+	}
+	parent.Set(expvarName, &g.expvar)
+	return g
+}
+
+func (g *gaugeExpvar) set(v float64) {
+	g.gauge.Set(v)
+	g.expvar.Set(int64(v))
+}
 
 var (
-	removalPolicyExpvar               = expvar.Map{}
-	newRemovalPolicyCountExpvar       = expvar.Int{}
-	registeredDomainCountExpvar       = expvar.Int{}
-	outdatedFilesCountExpvar          = expvar.Int{}
-	filesFromUnknownDomainCountExpvar = expvar.Int{}
-
-	transactionContainerExpvar     = expvar.Map{}
-	currentMemSizeInBytesExpvar    = expvar.Int{}
-	transactionsCountExpvar        = expvar.Int{}
-	transactionsDroppedCountExpvar = expvar.Int{}
-	errorsCountExpvar              = expvar.Int{}
-
-	fileStorageExpvar                  = expvar.Map{}
-	serializeCountExpvar               = expvar.Int{}
-	deserializeCountExpvar             = expvar.Int{}
-	fileSizeExpvar                     = expvar.Int{}
-	currentSizeInBytesExpvar           = expvar.Int{}
-	filesCountExpvar                   = expvar.Int{}
-	reloadedRetryFilesCountExpvar      = expvar.Int{}
-	filesRemovedCountExpvar            = expvar.Int{}
-	deserializeErrorsCountExpvar       = expvar.Int{}
-	deserializeTransactionsCountExpvar = expvar.Int{}
+	removalPolicyExpvar                  = expvar.Map{}
+	newRemovalPolicyCountTelemetry       *counterExpvar
+	registeredDomainCountTelemetry       *counterExpvar
+	outdatedFilesCountTelemetry          *counterExpvar
+	filesFromUnknownDomainCountTelemetry *counterExpvar
+
+	transactionContainerExpvar        = expvar.Map{}
+	currentMemSizeInBytesTelemetry    *gaugeExpvar
+	transactionsCountTelemetry        *gaugeExpvar
+	transactionsDroppedCountTelemetry *counterExpvar
+	errorsCountTelemetry              *counterExpvar
+
+	fileStorageExpvar                     = expvar.Map{}
+	serializeCountTelemetry               *counterExpvar
+	deserializeCountTelemetry             *counterExpvar
+	fileSizeTelemetry                     *gaugeExpvar
+	currentSizeInBytesTelemetry           *gaugeExpvar
+	filesCountTelemetry                   *gaugeExpvar
+	reloadedRetryFilesCountTelemetry      *counterExpvar
+	filesRemovedCountTelemetry            *counterExpvar
+	deserializeErrorsCountTelemetry       *counterExpvar
+	deserializeTransactionsCountTelemetry *counterExpvar
 )
 
 func init() {
 	forwarderExpvars.Set("RemovalPolicy", &removalPolicyExpvar)
-	removalPolicyExpvar.Set("NewRemovalPolicyCount", &newRemovalPolicyCountExpvar)
-	removalPolicyExpvar.Set("RegisteredDomainCount", &registeredDomainCountExpvar)
-	removalPolicyExpvar.Set("OutdatedFilesCount", &outdatedFilesCountExpvar)
-	removalPolicyExpvar.Set("FilesFromUnknownDomainCount", &filesFromUnknownDomainCountExpvar)
+	newRemovalPolicyCountTelemetry = newCounterExpvar(
```

Comment:
Seems like `expvar_name` is just a camel case version of name. Could we compute it in `newGaugeExpvar` and `newCounterExpvar` ?

Reply:
When I did this development, I was wondering if I should compute the name of the expvar from the name of the telemetry. 
I think telemetry name and expvar name might be different but as it is the same now, it is probably a good idea to compute `expvarName` from `name`.